Cython/Compiler/Buffer.py

```diff
@@ -679,8 +679,12 @@ def get_type_information_cname(code, dtype, maxdepth=None):
                      for f in fields]
             typecode.putln("static __Pyx_StructField %s[] = {" % structinfo_name, safe=True)
             for f, typeinfo in zip(fields, types):
+                if dtype.needs_cpp_construction:
```

Comment:
Is this the optimal way to check that the piece of code we will be adding is in cpp?

Reply:
Otherwise a lot of windows cpp tests fail

Reply:
the deal is `offsetof(struct Foo, member_name)` is ok for c, but not as ok for cpp, so msvc don't allow this. So i have to make it `offsetof(Foo, member_name)

Reply:
> I'd rather expect a change like this in a separate PR.

Well, if you want .. but it seems easier to fix failing tests right here, in the place where i add them

Reply:
> but it seems easier to fix failing tests right here, in the place where i add them

You do not think so?

Reply:
> > I'd rather expect a change like this in a separate PR.
> 
> Well, if you want .. but it seems easier to fix failing tests right here, in the place where i add them

The problem is not new in this PR, is it? It's already there, and appveyor probably isn't happy with it either. Making it a part of this PR means that it will be buried under a heap of CI config related changes in the git history and for later generations of developers (and my future self and maybe yours), it will be difficult to understand why this was done at the time. Let's be nice to the people who live in the future.

Reply:
Ok, I'll I will put fixes in a separate PR

Reply:
Or maybe seceral separate PR's, one for each problem?

Reply:
here #4331